File: skellig-connection-database/src/main/java/org/skellig/connection/database/model/DatabaseRequest.java
package org.skellig.connection.database.model;

import java.util.Map;
import java.util.Optional;

public class DatabaseRequest {

    private String command;
    private String table;
    private String query;
    private Map<String, Object> columnValuePairs;

    public DatabaseRequest(String command, String table, Map<String, Object> columnValuePairs) {
        this.command = command;
        this.table = table;
        this.columnValuePairs = columnValuePairs;
    }

    public DatabaseRequest(String query) {
        this.query = query;
    }

    public String getCommand() {
        return command;
    }

    public String getTable() {
        return table;
    }

    public Optional<String> getQuery() {
        return Optional.ofNullable(query);
    }

    public Optional<Map<String, Object>> getColumnValuePairs() {
        return Optional.ofNullable(columnValuePairs);
    }
}

File: skellig-test-step-processing-db/src/main/kotlin/org/skellig/teststep/processor/db/model/DatabaseRequest.kt
package org.skellig.teststep.processor.db.model

/**
 * This class represents a Database Request.
 *
 * A Database Request can be used to create database queries or commands.
 *
 * The properties [command], [table] and [columnValuePairs] must be defined together or [query] with [queryParameters].
 *
 * @constructor Creates a DatabaseRequest object with the specified SQL query and query parameters.
 * @param query SQL query string.
 * @param queryParameters The list of query parameters.
 *
 * @constructor Creates a DatabaseRequest object with the specified command, table, and column-value pairs.
 * @param command The command string.
 * @param table The table name.
 * @param columnValuePairs The map of column-value pairs.
 */
open class DatabaseRequest private constructor(val query: String? = null,
                                               val queryParameters: List<Any?>? = null,
                                               val command: String? = null,
                                               val table: String? = null,
                                               val columnValuePairs: Map<String, Any?>? = null) {

    constructor(query: String?, queryParameters: List<Any?>? = null) : this(query, queryParameters, null, null, null)

    constructor(command: String? = null,
                table: String? = null,
                columnValuePairs: Map<String, Any?>? = null) : this(null, null, command, table, columnValuePairs)

}